src/sqlancer/doris/DorisSchema.java

```diff
@@ -57,12 +57,31 @@ public enum DorisDataType {
         private int decimalPrecision;
         private int varcharLength;
 
-        public static DorisDataType getRandomWithoutNull() {
-            DorisDataType dt;
-            do {
-                dt = Randomly.fromOptions(values());
-            } while (dt == DorisDataType.NULL);
-            return dt;
+        public static DorisDataType getRandomWithoutNull(DorisOptions options) {
```

Comment:
I'd actually suggest including all types by default, and removing them when an option is specified. With the current version, if someone adds a new type, they need to debug why it is not generated.

Reply:
I'd suggest that we merge the PR after fixing this, so the CI tests run automatically, making it easier to fix them.

Reply:
> I'd suggest that we merge the PR after fixing this, so the CI tests run automatically, making it easier to fix them.

Yes, I'm working on fixing CI. But I'm only free on weekends, so it might be a bit slow.

Reply:
> I'd actually suggest including all types by default, and removing them when an option is specified. With the current version, if someone adds a new type, they need to debug why it is not generated.

I think this can be divided into two functions, one is to generate all basic types without parameters, and the other is to generate legal types based on options.